File: 3-SAT/3-sat/3-sat-core/src/main/java/com/anjlab/sat3/ICompactTripletsStructure.java
package com.anjlab.sat3;

public interface ICompactTripletsStructure extends ITabularFormula
{
    ICompactTripletsStructure subtract(ITabularFormula formula);
    ICompactTripletsStructure union(ICompactTripletsStructure cts);
    ICompactTripletsStructure intersect(ICompactTripletsStructure cts);
    ICompactTripletsStructure concretize(int varName, boolean value);
}

File: 3-SAT/FormulaCreator/ICompactTripletsStructure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FormulaCreator
{
    public interface ICompactTripletsStructure : ITabularFormula
    {
        ICompactTripletsStructure Subtract(ITabularFormula formula);
        ICompactTripletsStructure Union(ICompactTripletsStructure cts);
        ICompactTripletsStructure Intersect(ICompactTripletsStructure cts);
        ICompactTripletsStructure Concretize(int varName, bool value);
    }
}
